src/package-linker.js

```diff
@@ -271,7 +271,18 @@ export default class PackageLinker {
     // assume it's a link:-managed dependency, and overwrite it as usual.
     const linkTargets = new Map();
 
-    for (const entry of await fs.readdir(this.config.linkFolder)) {
+    let linkedModules;
+    try {
+      linkedModules = await fs.readdir(this.config.linkFolder);
+    } catch (err) {
+      if (err.code === 'ENOENT') {
+        linkedModules = [];
+      } else {
+        throw err;
+      }
+    }
+
+    for (const entry of linkedModules) {
```

Comment:
This code looks redundant with the one in config.js - can't we use `this.linkedModules`?

Reply:
Good catch! I think we can consolidate both code paths but I don't think it is as easy as it looks since there's some repeated logic with branching. I'll try to do it in a follow-up patch.